### PR TITLE
Add fast_geglu for Stable Diffusion

### PR DESCRIPTION
**Pull request by yuantailing**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Add `fast_geglu`, a torch extension, to speed up Stable Diffusion training. The extension is faster than the torch.compile.

**Collection**: stable_diffusion

# Changelog

- Add fast_geglu for Stable Diffusion

# Usage

- You can potentially add a usage example below

```python
import torch
from nemo.collections.multimodal.modules.stable_diffusion import fast_geglu

x_and_gate = torch.rand([8, 1024, 5120], dtype=torch.float16, device="cuda").requires_grad_()
output = fast_geglu.geglu(x_and_gate)
grad_output = torch.rand([8, 1024, 2560], dtype=torch.float16, device="cuda")
output.backward(grad_output)
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by alpha0422** [comment]
Hi @Victor49152, could you help review this change?

File: nemo/collections/multimodal/modules/stable_diffusion/attention.py [hunk 1]
```diff
@@ -106,8 +107,12 @@ def __init__(self, dim_in, dim_out):
         self.proj = LinearWrapper(dim_in, dim_out * 2)
 
     def forward(self, x):
-        x, gate = self.proj(x).chunk(2, dim=-1)
-        return x * F.gelu(gate)
+        x = self.proj(x)
+        if x.is_cuda and x.dtype == torch.float16 and x.shape[-1] // 2 in [1280, 2560, 5120] and x.is_contiguous():
```
Comment (mmarcinkiewicz) [inline nemo/collections/multimodal/modules/stable_diffusion/attention.py:111]: Is there a way we can hide the logic here in `fast_geglu` somewhere? and if the conditions are not met, we fallback to `F.gelu()`?

Comment (yuantailing) [reply nemo/collections/multimodal/modules/stable_diffusion/attention.py]: Done

**Review by Victor49152 — APPROVED** [review]
LGTM
